wgsl/index.bs

```diff
@@ -5709,6 +5712,377 @@ Note: If the function [=return value|returns a value=], that value is ignored.
 </div>
 
 
+## Statements Behavior Analysis ## {#behaviors}
+
+### Rules ### {#behaviors-rules}
+
+Some statements affecting control-flow are only valid in some contexts.
+For example, `fallthrough` is invalid outside of a switch, and `continue` is invalid outside of a loop.
+Additionally, the uniformity analysis (see [[#uniformity]]) needs to know when control flow can exit a statement in multiple different ways.
+
+Both goals are achieved by a system for summarizing execution behaviors of statements and expressions. Behavior analysis maps each statement and expression to the set of possible ways execution proceeds after evaluation of the statement or expression completes.
+As with type analysis for values and expressions, behavior analysis proceeds bottom up: first determine behaviors for certain basic statements, and then determine behavior for higher level constructs by applying combining rules.
+
+A <dfn export>behavior</dfn> is a set, whose elements may be:
+- Return
+- Discard
+- Break
+- Continue
+- Fallthrough
+- Next
+
+Each of those correspond to a way to exit a compound statement: either through a keyword, or by falling to the next statement ("Next").
+
+We note "*s*: *B*" to say that *s* respects the rules regarding behaviors, and has [=behavior=] *B*.
+
+For each function:
+- its body must be a valid statement by these rules
+- if it has a return type, its [=behavior=] must be one of {Return} or {Return, Discard}
+- otherwise, its [=behavior=] must be a subset of {Next, Return, Discard}
+
+We assign a [=behavior=] to each function: it is its body's [=behavior=] (treating the body as a regular statement), with any "Return" replaced by "Next".
+As a consequence of the rules above, a function behavior is always one of {Next}, {Discard} or {Next, Discard}.
```

Comment:
I had missed the {} case, which can happen when a function just contains an infinite loop with no way to exit. I fixed it.